xbmc/Application.cpp

```diff
@@ -4211,9 +4213,9 @@ bool CApplication::OnMessage(CGUIMessage& message)
         CGUIMessage msg(GUI_MSG_PLAYLISTPLAYER_CHANGED, 0, 0, g_playlistPlayer.GetCurrentPlaylist(), param, item);
         g_windowManager.SendThreadMessage(msg);
         g_playlistPlayer.SetCurrentSong(m_nextPlaylistItem);
-        *m_itemCurrentFile = *item;
+        m_itemCurrentFile = item;
```

Comment:
ok, I'll change it but you should fix this issue as well. if you don't want the object to be manipulated outside, don't give pointers to the outside. a client should not bother with this.